packages/commonwealth/client/scripts/state/api/comments/editComment.ts

```diff
@@ -30,50 +30,56 @@ const editComment = async ({
     thread_id: threadId,
     body: updatedBody,
     parent_comment_id: parentCommentId,
-  })
+  });
 
-  const response = await axios.patch(`${app.serverUrl()}/comments/${commentId}`, {
-    address: address,
-    author_chain: chainId,
-    id: commentId,
-    chain: chainId,
-    body: encodeURIComponent(updatedBody),
-    jwt: app.user.jwt,
-    canvas_action: action,
-    canvas_session: session,
-    canvas_hash: hash,
-  })
+  const response = await axios.patch(
+    `${app.serverUrl()}/comments/${commentId}`,
+    {
+      address: address,
+      author_chain: chainId,
+      id: commentId,
+      chain: chainId,
+      body: encodeURIComponent(updatedBody),
+      jwt: app.user.jwt,
+      canvas_action: action,
+      canvas_session: session,
+      canvas_hash: hash,
+    }
+  );
 
-  return new Comment(response.data.result)
+  return new Comment(response.data.result);
 };
 
 interface UseEditCommentMutationProps {
-  chainId: string
+  chainId: string;
   threadId: number;
 }
 
-const useEditCommentMutation = ({ chainId, threadId }: UseEditCommentMutationProps) => {
+const useEditCommentMutation = ({
+  chainId,
+  threadId,
+}: UseEditCommentMutationProps) => {
   const queryClient = useQueryClient();
   const { data: comments } = useFetchCommentsQuery({
     chainId,
     threadId,
-  })
+  });
 
   return useMutation({
     mutationFn: editComment,
     onSuccess: async (updatedComment) => {
       // update fetch comments query state with updated comment
-      const key = [ApiEndpoints.FETCH_COMMENTS, chainId, threadId]
+      const key = [ApiEndpoints.FETCH_COMMENTS, chainId, threadId];
       queryClient.cancelQueries({ queryKey: key });
-      queryClient.setQueryData([...key],
-        () => {
-          // find the existing comment index, and return updated comment in its place
-          return comments.map(x => x.id === updatedComment.id ? updatedComment : x);
-        }
-      );
+      queryClient.setQueryData([...key], () => {
+        // find the existing comment index, and return updated comment in its place
```

Comment:
In general I feel like we use the same set of mutation patterns in these `setQueryData` calls:
1. Add new item to array
2. Delete item from array
3. Edit item in array

We may want to encapsulate those generally in a querying utility, which I believe is what @masvelio was asking for with the array destructuring.

Reply:
Hmm, generally we won't be always doing those actions based on Id. Also, most objects won't be the same I assume.

I believe we can made a dedicated util per entity (thread/comments). Do you have any suggestions for how the querying utility should look like?

Reply:
Out of scope for this PR, in my opinion, but let's take some time to review / research it and we can come up with a good helper. Feel free to make a quick ticket to investigate setQueryData utils.